src/layer/AbstractLayer.ts

```diff
@@ -60,21 +58,103 @@ export class AbstractLayer {
     if (!this.dataset || !this.dataset.orbitTimeMinutes) {
       throw new Error('Orbit time is needed for grouping tiles into flyovers.');
     }
-
     if (bbox.crs !== CRS_EPSG4326) {
       throw new Error('Currently, only EPSG:4326 in findFlyovers');
     }
 
-    let tiles: Tile[] = [];
+    const orbitTimeMS = this.dataset.orbitTimeMinutes * 60 * 1000;
+    const bboxGeometry: Geom = this.roundCoordinates([
+      [
+        [bbox.minX, bbox.minY],
+        [bbox.maxX, bbox.minY],
+        [bbox.maxX, bbox.maxY],
+        [bbox.minX, bbox.maxY],
+        [bbox.minX, bbox.minY],
+      ],
+    ]);
+    const bboxArea = area({
+      type: 'Polygon',
+      coordinates: bboxGeometry,
+    });
+
+    let flyovers: FlyoverInterval[] = [];
+    let flyoverIndex = 0;
+    let currentFlyoverGeometry: Geom = null;
+    let nTilesInFlyover;
+    let sumCloudCoverPercent;
     for (let i = 0; i < maxFindTilesRequests; i++) {
-      const { tiles: partialTiles, hasMore } = await this.findTiles(
+      // grab new batch of tiles:
+      const { tiles, hasMore } = await this.findTiles(
```

Comment:
This algorithm  assumes tiles will always be sorted descending ( from most to least recent). Perhaps some parameter could be added to findTiles to make this more explicit. No need to do it in scope of this MR.

Reply:
Yes, agree completely, but would prefer to solve this in a future MR.